backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/service/CardActivator.kt

```diff
@@ -16,6 +19,8 @@ object CardActivator {
         val keyGenerator = KeyGenerator.getInstance(algorithm)
         keyGenerator.init(TOTP_SECRET_LENGTH * 8)
 
-        return keyGenerator.generateKey().encoded
+        return hashKey(keyGenerator.generateKey().encoded)
```

Comment:
I think we should not hash the TOTP secret

We should only hash the activation secret when writing to DB, and use the `Bcrypt.verifyer().verify(rawActiviationSecret, activationSecretHash)` method when a user is trying to activate his card.

(My explanation may be confusing -- I'm a bit sluggish today^^)

Reply:
Got you! I was kind of wondering whether I did this correct in first place.